Push crd-annotations to Maven central

Using the Strimzi `api` module in an application for handling Kafka custom resource, if the dependency-versions-check-maven-plugin is used, the build fails with this error:

```
[ERROR] Failed to execute goal org.basepom.maven:dependency-versions-check-maven-plugin:3.2.0:check (basepom.default) on project kas-fleetshard-operator: Could not resolve dependencies: [io.strimzi:crd-annotations:jar:0.22.0 [compile]] -> [Help 1]
```

We noticed that Maven central has `api` and `crd-generator` jars but not the `crd-annotations` one which is a transitive dependency. Maybe it's worth pushing it to Maven central as well?

--- a/.azure/scripts/push-to-nexus.sh
+++ b/.azure/scripts/push-to-nexus.sh
@@ -5,7 +5,7 @@ export GPG_TTY=$(tty)
 echo $GPG_SIGNING_KEY | base64 -d > signing.gpg
 gpg --batch --import signing.gpg
 
-GPG_EXECUTABLE=gpg mvn -B $MVN_ARGS -DskipTests -s ./.azure/scripts/settings.xml -pl ./,crd-generator,crd-annotations,api,api-conversion,test-container -P ossrh deploy
+GPG_EXECUTABLE=gpg mvn -B $MVN_ARGS -DskipTests -s ./.azure/scripts/settings.xml -pl ./,crd-annotations,crd-generator,api,api-conversion,test-container -P ossrh deploy
 
 rm -rf signing.gpg
 gpg --delete-keys

--- a/.azure/scripts/push-to-nexus.sh
+++ b/.azure/scripts/push-to-nexus.sh
@@ -5,7 +5,7 @@ export GPG_TTY=$(tty)
 echo $GPG_SIGNING_KEY | base64 -d > signing.gpg
 gpg --batch --import signing.gpg
 
-GPG_EXECUTABLE=gpg mvn -B $MVN_ARGS -DskipTests -s ./.azure/scripts/settings.xml -pl ./,crd-generator,api,api-conversion,test-container -P ossrh deploy
+GPG_EXECUTABLE=gpg mvn -B $MVN_ARGS -DskipTests -s ./.azure/scripts/settings.xml -pl ./,crd-generator,crd-annotations,api,api-conversion,test-container -P ossrh deploy
 
 rm -rf signing.gpg
 gpg --delete-keys

--- a/api/pom.xml
+++ b/api/pom.xml
@@ -87,6 +87,7 @@
         <dependency>
             <groupId>io.strimzi</groupId>
             <artifactId>test</artifactId>
+            <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>io.strimzi</groupId>

